petastorm/gcsfs_helpers/gcsfs_wrapper.py

```diff
@@ -41,26 +38,23 @@ def walk(self, path):
         directories = set()
         files = set()
 
-        for key in self.fs.ls(path, detail=True):
+        for obj in self.fs.ls(path, detail=True):
             # each info name must be at least [path]/part , but here
             # we check also for names like [path]/part/
-            path = key['name']
-            if key['storageClass'] == 'DIRECTORY':
-                if path.endswith('/'):
-                    directories.add(path[:-1])
-                else:
-                    directories.add(path)
-            elif key['storageClass'] == 'BUCKET':
-                pass
-            else:
-                files.add(path)
-
-        files = sorted([posixpath.split(f)[1] for f in files
-                        if f not in directories])
-        directories = sorted([posixpath.split(x)[1]
-                              for x in directories])
-
-        yield path, directories, files
+            obj_path = obj['name']
+            if obj_path == path:
```

Comment:
This check still fails to catch the desired directory. `obj_path` has an extra '/' on the end which `path` does not.

Reply:
Thanks, I was able to reproduce, but for some reason only when I provided a directory that didn't contain files, only other directories.